frontend/docker-compose-e2e-tests.yml

```diff
@@ -16,24 +16,45 @@ services:
     build:
       context: ../
       dockerfile: Dockerfile
+      x-bake:
+        cache-from:
+          - type=gha,scope=image-e2e-unified-api
+        cache-to:
+          - type=gha,mode=max,scope=image-e2e-unified-api
+        tags:
+          - frontend_flagsmith-api
     environment:
       E2E_TEST_AUTH_TOKEN: some-token
-      FE_E2E_TEST_USER_EMAIL: nightwatch@solidstategroup.com
+      ENABLE_FE_E2E: 'True'
       DJANGO_ALLOWED_HOSTS: '*'
       DATABASE_URL: postgresql://postgres:password@db:5432/flagsmith
       DISABLE_ANALYTICS_FEATURES: 'true'
       EMAIL_BACKEND: django.core.mail.backends.smtp.EmailBackend
+      ACCESS_LOG_LOCATION: /dev/shm/log.txt
     ports:
       - 8000:8000
     depends_on:
       - db
     links:
       - db:db
+    healthcheck:
+      test: "[ -e /dev/shm/log.txt ] && exit 0 || exit 1"
+      start_period: 60s
+      interval: 10s
+      timeout: 3s
+      retries: 30
```

Comment:
Sorry, I really don't understand what we're gaining here? What was wrong with the previous approach?

Reply:
The previous approach, locally in my environment at least, doesn't wait for the API to be up and running, so it fails since migrations take some time and E2E tests start before the API service is ready.
![image](https://github.com/Flagsmith/flagsmith/assets/41410593/9919092b-2d1b-459f-b30d-47f05d3a3c04)
https://docs.docker.com/compose/startup-order/

Reply:
Ok 👍